### DISCUSSION TITLE
no point data read

### DISCUSSION THREAD

**Discussion by wentma** [pyvista/pyvista#3785] [Q&A]
Hello,

I'm trying to read in a .vtu dataset with:

```python
reader = pv.get_reader(file)
mesh = reader.read() 
```

the mesh is read correctly however the point data are not read. With the pyvista version 0.31.0 it was possible to read the complete data set correctly with:

```python
mesh = pv.read(file)
```

I notices, that the error already occurs with version 0.32.0. Am I missing something in the newer versions?

Best regards

file:
[elipsoide_vis01-0037.zip](https://github.com/pyvista/pyvista/files/10359136/elipsoide_vis01-0037.zip)

**Comment by banesullivan** [comment]
It appears there are no point data in this file?...

The file contents have:

```xml
   <CellData Scalars="scalars">
    <DataArray type="Float32" Name="Pressure" Format="appended" offset="78136"/>
   </CellData>
```

and that's it. 

When I read this file, I get the Pressure CELL data array and no point data. 

This seems expected.

**Reply by MatthewFlamm** [reply]
It is _possible_ that when I cleaned up the old `pv.read` functionality we removed a `cell_data_to_point_data` filter call?  I'm sorry if this happened and it wasn't documented well enough.  If this is what occurred, you can do:

```py
reader = pv.get_reader(file)
mesh = reader.read().cell_data_to_point_data()  # use `pass_cell_data=True` if you want both
```

Edit:  I looked through those changes and couldn't find an obvious place where this functionality would have been removed, so I would also double check the files being used for point data.

**Reply by MatthewFlamm** [reply]
As an aside that might help debug this, a plug for the power of the Reader classes, you can also inspect the data before it is read:

```py
import pyvista as pv
reader = pv.get_reader("elipsoide_vis01-0037.vtu")
reader.cell_array_names  # ['Pressure']
reader.point_array_names  # []
```

**Reply by wentma** [reply]
Thank you for your help! I think I know now what went wrong. I was expecting, that the reader automatically sets the array as active scalar (as it seems to be done by the older versions). Therefore I only looked into the active scalar which is now empty.

Best regards

**Reply by MatthewFlamm** [reply]
Now this could be a regression.  Thanks for the report.  I'd have to go back to the original code, but I'm not sure how we could decide which scalars should be active if there are multiple point or cell data arrays.
